File: products/LoanHub/apps/frontend/types/companyRegistration.ts
import type { Gender, MaritalStatus } from "@/types/person";

export type InstitutionType =
    | "loan_company"
    | "commercial_bank"
    | "microfinance_institution"
    | "financial_cooperative"
    | "development_finance_institution"
    | "government_lending_program";

export type CompanyOwnerRegistrationPayload = {
    institution_type?: InstitutionType;
    company_name: string;
    registration_number?: string | null;
    license_number?: string | null;
    company_phone: string;
    company_email?: string | null;
    website?: string | null;
    address?: string | null;
    district?: string | null;
    owner_email?: string | null;
    owner_phone: string;
    password: string;
    first_name: string;
    middle_name?: string | null;
    last_name: string;
    gender?: Gender | null;
    date_of_birth?: string | null;
    national_id?: string | null;
    passport_number?: string | null;
    marital_status?: MaritalStatus | null;
    nationality?: string | null;
    town_or_village?: string | null;
    physical_address?: string | null;
};

export type CompanyOwnerRegistrationResponse = {
    company_id: string;
    institution_type: InstitutionType;
    owner_user_id: string;
    staff_membership_id: string;
    status: string;
    message: string;
};

File: products/LoanHub/apps/backend/database/schemas/company_registration.py
from datetime import date
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, EmailStr, Field

from database.models.enums import Gender, InstitutionType, MaritalStatus


class CompanyOwnerRegistrationCreate(BaseModel):
    institution_type: InstitutionType = InstitutionType.LOAN_COMPANY
    company_name: str = Field(min_length=2, max_length=200)
    registration_number: Optional[str] = Field(default=None, max_length=100)
    license_number: Optional[str] = Field(default=None, max_length=100)
    company_phone: str = Field(min_length=8, max_length=30)
    company_email: Optional[EmailStr] = None
    website: Optional[str] = Field(default=None, max_length=200)
    address: Optional[str] = None
    district: Optional[str] = Field(default=None, max_length=100)

    owner_email: Optional[EmailStr] = None
    owner_phone: str = Field(min_length=8, max_length=30)
    password: str = Field(min_length=8, max_length=128)
    first_name: str = Field(min_length=1, max_length=100)
    middle_name: Optional[str] = Field(default=None, max_length=100)
    last_name: str = Field(min_length=1, max_length=100)
    gender: Optional[Gender] = None
    date_of_birth: Optional[date] = None
    national_id: Optional[str] = Field(default=None, max_length=50)
    passport_number: Optional[str] = Field(default=None, max_length=50)
    marital_status: Optional[MaritalStatus] = None
    nationality: Optional[str] = "Mosotho"
    town_or_village: Optional[str] = None
    physical_address: Optional[str] = None


class CompanyOwnerRegistrationRead(BaseModel):
    company_id: UUID
    institution_type: InstitutionType
    owner_user_id: UUID
    staff_membership_id: UUID
    status: str
    message: str
